Use build.cmd for integration tests

drewnoakes opened the pull request.

davkean commented:
This currently produces two builds for VS integration tests; Debug & Release. I would pick one (probably Debug for asserts) and run with it.

davkean commented:
Restore failed in the build, clearly this doesn't block build when it should:

```
F:\workspace\_work\1\s\build\Toolset.proj : error NU1604: Project dependency RoslynTools.VsixExpInstaller does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results.
F:\workspace\_work\1\s\build\Toolset.proj : error NU1103: Unable to find a stable package RoslynTools.VsixExpInstaller with version 
F:\workspace\_work\1\s\build\Toolset.proj : error NU1103:   - Found 419 version(s) in https://dotnet.myget.org/F/roslyn-tools/api/v3/index.json [ Nearest version: 1.0.0-beta-62320-02 ]
F:\workspace\_work\1\s\build\Toolset.proj : error NU1103:   - Found 0 version(s) in nuget.org
F:\workspace\_work\1\s\build\Toolset.proj : error NU1103:   - Found 0 version(s) in https://dotnet.myget.org/F/msbuild/api/v3/index.json
F:\workspace\_work\1\s\build\Toolset.proj : error NU1103:   - Found 0 version(s) in https://dotnet.myget.org/F/nuget-build/api/v3/index.json
F:\workspace\_work\1\s\build\Toolset.proj : error NU1103:   - Found 0 version(s) in https://dotnet.myget.org/F/project-system-interop/api/v3/index.json
F:\workspace\_work\1\s\build\Toolset.proj : error NU1103:   - Found 0 version(s) in https://dotnet.myget.org/F/roslyn-analyzers/api/v3/index.json
F:\workspace\_work\1\s\build\Toolset.proj : error NU1103:   - Found 0 version(s) in https://dotnet.myget.org/F/roslyn/api/v3/index.json
F:\workspace\_work\1\s\build\Toolset.proj : error NU1103:   - Found 0 version(s) in https://dotnet.myget.org/F/symreader-converter/api/v3/index.json
F:\workspace\_work\1\s\build\Toolset.proj : error NU1103:   - Found 0 version(s) in https://dotnet.myget.org/F/templating/api/v3/index.json
F:\workspace\_work\1\s\build\Toolset.proj : error NU1103:   - Found 0 version(s) in https://vside.myget.org/F/devcore/api/v3/index.json
F:\workspace\_work\1\s\build\Toolset.proj : error NU1103:   - Found 0 version(s) in https://vside.myget.org/F/vs-impl/api/v3/index.json
F:\workspace\_work\1\s\build\Toolset.proj : error NU1103:   - Found 0 version(s) in https://vside.myget.org/F/vssdk/api/v3/index.json
```

davkean commented:
I fixed the restore failure here: https://github.com/dotnet/project-system/pull/5114.

drewnoakes commented:
> I fixed the restore failure here: #5114. 

Nice. I've updated this PR.

> This currently produces two builds for VS integration tests; Debug & Release

Actually it appears to have three. Agree we should only have Debug. Will look at this tomorrow.

drewnoakes commented:
Appears to still be stuck on the same line (though the restore issue is fixed):

```
  Microsoft.VisualStudio.ProjectSystem.Managed.TestServices.UnitTests -> F:\workspace\_work\1\s\artifacts\Debug\bin\UnitTests\Microsoft.VisualStudio.ProjectSystem.Managed.TestServices.UnitTests.dll
  Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests -> F:\workspace\_work\1\s\artifacts\Debug\bin\UnitTests\Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests.dll
  Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests -> F:\workspace\_work\1\s\artifacts\Debug\bin\UnitTests\Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests.dll
  VisualStudioEditorsSetup -> F:\workspace\_work\1\s\artifacts\Debug\VSSetup\VisualStudioEditorsSetup.vsix
  Setting up Visual Studio for debugging extensions. This one-time operation may take a minute or more.
```

davkean commented:
I suspect its not stuck, just doing a bunch of work and its taking a long time.

drewnoakes commented on build/azure-pipelines-integration.yml:
```diff
@@ -15,11 +15,9 @@ jobs:
   timeoutInMinutes: 60
   steps:
-    - script: $(Build.SourcesDirectory)\build\IntegrationCIBuild.cmd -configuration $(_configuration) -prepareMachine -log
+    - script: $(Build.SourcesDirectory)\build.cmd /build /no-test /integration /diagnostic /ci /no-sign /no-deploy /configuration $(_configuration)
```
`/no-deploy` will need to be removed before this CI work is done.
